src/common.ts

```
@@ -514,3 +514,7 @@ export async function getArUSDPrice(): Promise<number> {
 		return 0;
 	}
 }
+
+export function winstonToAr(winston: number): number {
```

We should throw an exception if the absolute value of the Winston number is less than 1.0 since there's no such thing as fractional Winston; it's the smallest unit of AR.

---

For reference: https://docs.arweave.org/developers/server/http-api#ar-and-winston

---

Is it worth it here to throw an exception for this case or should we simply round the number to the nearest integer before attempting conversion? 

Having it round for us would give the function a little more utility and become easier to use outside of a try/catch block. Ideally, the Arweave gateway will only ever return us a whole winston number. So the only way we normally would end up with a decimal is from some type of conversion we made in the code after we have obtained that value from the gateway.

A decimal of a winston is such a small fraction I'm just not sure its "worth" throwing errors.

So @arielmelendez , do you think this function should be:

```ts
// With exception
export function winstonToAr(winston: number): number {
  if (Number.isInteger(winston)) throw new Error(`Winston value not an integer: ${winston}`);
  return winston * 0.000000000001;
}
```

Or:

```ts
// With rounding
export function winstonToAr(winston: number): number {
  return Math.round(winston) * 0.000000000001;
}
```

---

Yea we'll see how this goes. Thanks for the update.